File: java/src/Enseignant.java
public class Enseignant extends Personne {
    private double salaire;

    public Enseignant(String nom, int age, double salaire) {
        super(nom, age);
        setSalaire(salaire);
    }

    public double getSalaire() { return salaire; }

    public void setSalaire(double salaire) {
        if (salaire < 0)
            throw new IllegalArgumentException("Le salaire ne peut pas être négatif");
        this.salaire = salaire;
    }

    @Override
    public void afficherDetails() {
        System.out.println("Enseignant: " + getNom() + " | Salaire: " + salaire + " €");
    }

    @Override
    public String toString() {
        return "Enseignant: " + getNom() + ", " + getAge() + " ans | Salaire: " + salaire + " €";
    }
}

File: python/src/enseignant.py
from .personne import Personne

class Enseignant(Personne):
    def __init__(self, nom: str, age: int, salaire: float):
        super().__init__(nom, age)
        self.salaire = salaire

    @property
    def salaire(self) -> float:
        return self._salaire

    @salaire.setter
    def salaire(self, valeur: float):
        if valeur < 0:
            raise ValueError("Le salaire ne peut pas être négatif")
        self._salaire = valeur

    def afficher_details(self):
        print(f"Enseignant: {self.nom} | Salaire: {self.salaire} €")

    def __str__(self):
        return f"Enseignant: {self.nom}, {self.age} ans | Salaire: {self.salaire} €"
